### ISSUE TITLE
Suggestion: check_return_type()

### ISSUE THREAD

**Issue by mikenerone** [agronholm/typeguard#3]
For cases where a developer wishes to avoid the extra frame of the `@typechecked` decorator but would still like to check the return value, `typeguard` could provide a `check_return_type()` function to call directly before `return`ing, similar to `check_argument_types()`.

**Comment by agronholm** [comment]
How exactly would that work?

**Comment by agronholm** [comment]
`check_argument_types()` works because the arguments have already been passed and it's only a matter of fishing them out using the stack and the garbage collector. But how would you determine the return type of a function when it hasn't even returned yet?

**Comment by mikenerone** [comment]
Sorry, I didn't elaborate enough. What I meant is that the developer would at least have the option of providing the return value explicitly prior to returning:
```
def f(......):
    assert check_argument_types()  # Yes, convenient that it finds the args itself
    ... # calculate
    ... # calculate
    result = ...
    assert check_return_type(result)  # No way to do the same here, but at least it's trivial and gets the job done
    return result
```
The return annotation would of course be introspected out of the stack in the same way that the argument annotations are.

**Comment by agronholm** [comment]
Ok, that makes a bit more sense now. I'll think about it.

**Comment by agronholm** [comment]
This requires some code refactoring. `check_return_types()` needs the bound types of each type variable in the function, and those are computed during `check_argument_types()` and will not be available to `check_return_types()`.

**Comment by agronholm** [comment]
I've just released 2.1.0. Would you mind checking if the third type checking style using the new TypeChecker class suits you? I'm not really keen on adding `check_return_type()` to the mix.
